### DISCUSSION TITLE
Open points

### DISCUSSION THREAD

**Discussion by woernsn** [Goldenfreddy0703/plugin.audio.ytmusic.exp#4] [General]
Hi @Goldenfreddy0703,

I just took a look at the open points in the README.md

- support adaptive streaming
As I can see from the code, it is already in place.
If a video provides a link that is adaptive straming capable, this one is used automatically:
https://github.com/Goldenfreddy0703/plugin.audio.ytmusic.exp/blob/main/resources/lib/login.py#L127:L128
- add Android/Linux support
I'm not sure what is missing there.
I am working and debugging under Linux (Ubuntu) and everything seems to work fine.
- add playlist operations
Do you have any idea what is meant by that?
- implement login like Youtube Addon
This is hard one. YouTube Music does not support the YouTube API keys, I think.
I might try to play around with it but I guess, it's gonna be tough and I'm not sure if it is worth the effort.

What do you think?

**Comment by Goldenfreddy0703** [comment]
Hey there @woernsn 

On the Kodi Forum, [foreverguest](https://forum.kodi.tv/member.php?action=profile&uid=141192) developed a todo list in the future for this. Here is what i think about the todo list. 

- support adaptive streaming
Your right, it is already in place so we should remove that feature.
- add Android/Linux support
Ok so there was some stuff that was missing, back then i tried using linux and it was working but some parts of the menu were not which caused issues. For Android, for some reason, you could not play any media because i think it was not reading the headers.json file properly which is why we need to fix that soon.
- add playlist operations
Honesty, I kinda had no idea myself on what this would be for, [foreverguest](https://forum.kodi.tv/member.php?action=profile&uid=141192) put this on his todo list but i did not really think it was needed. 
- implement login like Youtube Addon
[foreverguest](https://forum.kodi.tv/member.php?action=profile&uid=141192) also put this on his todo list cause logging in and getting access to other stuff without using a headers file would be useful to have. It is hard but if we can implant a login and use this github repo: [ytmusicapi](https://github.com/sigma67/ytmusicapi), Maybe it is possible to implement a login.

**Reply by woernsn** [reply]
@Goldenfreddy0703, by the way, I've sent you a Discord request to get in contact :wink:.

**Reply by Goldenfreddy0703** [reply]
Oh yeah..sorry, last weekend my discord account got hacked due not by links but some discord server so basically it's disabled.

**Reply by Goldenfreddy0703** [reply]
I might make a new one tomorrow tho, reason why im not making one today is cause its Monday and Monday is the worst.

**Reply by woernsn** [reply]
> I might make a new one tomorrow tho, reason why im not making one today is cause its Monday and Monday is the worst.

Alright.
You can add me anytime: woernsn#3656

**Reply by Goldenfreddy0703** [reply]
Nice, will be glad to. Thank you
